lib/oidc-client.js

```diff
@@ -390,6 +411,30 @@ OidcClient.prototype = {
 
     });
   },
+  
+  getTokenFromCodeAsync: function (code) {
+    utility.log("OidcClient.getTokenFromCodeAsync");
+    var self = this;
+    var settings = self._settings;
+    
+    return self.loadTokenEndpoint().then(function (token_endpoint) {
+      var config = {
+        url: token_endpoint,
+        json: true,
+        form: {
+          grant_type: 'authorization_code',
+          code: code,
+          redirect_uri: settings.redirect_uri,
+          client_id: settings.client_id,
+          client_secret: settings.client_secret
+        }
+      };
+      
+      utility.copy(self._settings.httpSettings, config);
+
+      return rp.post(config);
```

Comment:
this block should be simplified to not require RP at this level and go through utilites